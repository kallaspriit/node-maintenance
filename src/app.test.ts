import * as HttpStatus from "http-status-codes";
import * as supertest from "supertest";
import setupApp from "./app";

let app: supertest.SuperTest<supertest.Test>;

describe("app", () => {
  beforeEach(async () => {
    app = supertest.agent(await setupApp());
  });

  it("should render maintenance view for index", async () => {
    const response = await app.get("/").send();

    expect(response.status).toEqual(HttpStatus.OK);
    expect(response.text).toMatchSnapshot();
  });

  it("should render maintenance view for every url", async () => {
    const response = await app.get("/foo/bar").send();

    expect(response.status).toEqual(HttpStatus.OK);
    expect(response.text).toMatchSnapshot();
  });
});
