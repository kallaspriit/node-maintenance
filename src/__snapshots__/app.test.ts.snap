// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render maintenance view for every url 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Under Maintenance</title>
  </head>
  <body>
    <h1>Under Maintenance</h1>
    <p>Sorry for the inconvenience but the site is currently under planned maintenance.</p>
    <p>We will be back shortly, please check again in a few minutes.</p>
    <p><a href=\\"/foo/bar\\">Reload page</a></p>
  </body>
</html>"
`;

exports[`app should render maintenance view for index 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Under Maintenance</title>
  </head>
  <body>
    <h1>Under Maintenance</h1>
    <p>Sorry for the inconvenience but the site is currently under planned maintenance.</p>
    <p>We will be back shortly, please check again in a few minutes.</p>
    <p><a href=\\"/\\">Reload page</a></p>
  </body>
</html>"
`;
